libsolidity/inlineasm/AsmParser.cpp

```diff
@@ -252,7 +252,7 @@ assembly::Expression Parser::parseExpression()
 	{
 		Instruction const& instr = boost::get<Instruction>(operation);
 		int args = instructionInfo(instr.instruction).args;
-		if (args > 0 && currentToken() != Token::LParen)
+		if ((m_flavour != AsmFlavour::Loose || args > 0) && currentToken() != Token::LParen)
```

Comment:
This extra check here is actually not needed, but if you really want it I'd put it 4 lines above into the other condition.

Reply:
Sorry, which extra check? This thing should behave as follows:

If we are in loose mode, no `(` is fine if the instruction does not take arguments.
In strict mode, `(` is required in any case.